iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -381,9 +381,39 @@ public DecomposedPredicate decomposePredicate(JobConf jobConf, Deserializer dese
       }
     }
     predicate.pushedPredicate = (ExprNodeGenericFuncDesc) pushedPredicate;
+    setFilterExpression(predicate);
     return predicate;
   }
 
+  /**
+   * Sets the filter expression in the configuration.
+   *
+   * @param predicate The predicate to create the filter expression from.
+   */
+  public void setFilterExpression(DecomposedPredicate predicate) {
+    Expression expr = Expressions.alwaysTrue();
+    try {
+      if (predicate.pushedPredicate != null) {
+        SearchArgument sarg = ConvertAstToSearchArg.create(conf, predicate.pushedPredicate);
+        expr = HiveIcebergFilterFactory.generateFilterExpression(sarg);
```

Comment:
why don't we set the filter inside `HiveIcebergFilterFactory.generateFilterExpression` when it's invoked

Reply:
`HiveIcebergFilterFactory.generateFilterExpression ` is invoked  in multiple places. https://github.com/search?q=repo%3Aapache%2Fhive%20HiveIcebergFilterFactory.generateFilterExpression&type=code 

We just want this to be set during decomposePredicate.

Reply:
how about `IcebergInputFormat#applyConfig`?
````
if (filter != null) {
    setFilterExpr(filter)
}
....
private static setFilterExpr(Expression filter) {
    SessionStateUtil.addResource(conf, InputFormatConfig.QUERY_FILTERS, filter);
    scan = scan.filter(filter);
}
````